test/PackageManagement.Test/BuildIntegratedRestoreUtilityTests.cs

```diff
@@ -74,5 +620,22 @@ private static JObject BasicConfig
                 return json;
             }
         }
+
+
+        private class TestBuildIntegratedNuGetProject : BuildIntegratedNuGetProject
+        {
+            public IReadOnlyList<BuildIntegratedProjectReference> ProjectClosure { get; set;}
+
+            public TestBuildIntegratedNuGetProject(string jsonConfig, IMSBuildNuGetProjectSystem msbuildProjectSystem)
+                : base(jsonConfig, msbuildProjectSystem)
+            {
+                InternalMetadata.Add(NuGetProjectMetadataKeys.UniqueName, msbuildProjectSystem.ProjectName);
```

Comment:
To be safe, I will just add InternalMetadata.Add(NuGetProjectMetadataKeys.Name, msbuildProjectSystem.ProjectName); as well. Because, most of the code assumes that Name is always available